### PR TITLE
Add sub operator for Arm backend

### PR DESCRIPTION
**Pull request by Erik-Lundell**
Implemented node visitor, quantizer, and test.
TOSA MI and BI passes, U55 BI fails (expectedly) on Vela compilation.

Refactored code shared between sub and add.

Change-Id: Ifc9fc4ae083f3ed868ad763e4301e5fe87468a25

### CODE REVIEW

File: backends/arm/operators/op_sub.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        tosa_graph: ts.TosaSerializer,
+        inputs: List[TosaArg],
+        output: TosaArg,
+        is_quant_node: bool,
+        permute_memory_to_nhwc: bool,
+    ) -> None:
+        if is_quant_node:
+            input_nodes = tutils.get_two_inputs(node)
+
+            # Rescale inputs to 32 bit
```
Comment (digantdesai) [inline backends/arm/operators/op_sub.py:40]: Can you remind me again why this was needed? Can this be done with int16 for instance?

File: backends/arm/test/ops/test_sub.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+                module,
+                example_inputs=test_data,
+                compile_spec=common.get_u55_compile_spec(),
+            )
+            .quantize()
+            .export()
+            .check_count({"torch.ops.aten.sub.Tensor": 1})
+            .check(["torch.ops.quantized_decomposed"])
+            .to_edge()
+            .partition()
```
Comment (digantdesai) [inline backends/arm/test/ops/test_sub.py:92]: nit: do you want to put check_not after each partition?

File: backends/arm/tosa_quant_utils.py [hunk 3]
```diff
@@ -204,6 +205,68 @@ def build_rescale_from_int32(
...
+                tosa_graph,
+                tensor,
+                qarg.zp,
+                scale,
+            )
+        )
+    return rescaled_nodes, min_scale
+
+
+def rescale_node_back_to_int8(
```
Comment (digantdesai) [inline backends/arm/tosa_quant_utils.py:244]: I guess since we support only one quant dtype and quant schema it is ok to hardcode?

**Review by digantdesai — APPROVED** [review]

File: backends/arm/operators/op_sub.py (hunk 1 above)
Comment (Erik-Lundell) [reply backends/arm/operators/op_sub.py:40]: See supported dtypes here: https://www.mlplatform.org/tosa/tosa_spec.html#_sub
A quantized model is targeting BI for which only dtype(in_out_t) == i32_t is supported.

Currently, the implementation assumes that a quantized input tensor is int8 but we are moving away from that. Eventually.

File: backends/arm/tosa_quant_utils.py (hunk 3 above)
Comment (Erik-Lundell) [reply backends/arm/tosa_quant_utils.py:244]: See, above comment, I'd say that it's ok for now.

File: backends/arm/test/ops/test_sub.py (hunk 2 above)
Comment (Erik-Lundell) [reply backends/arm/test/ops/test_sub.py:92]: Good point, I'll keep it in mind as I go forward with new ops :)

**Comment by digantdesai** [comment]
Rebase please?

**Comment by digantdesai** [comment]
`test-arm-reference-delegation` job is still failing but looks conv related? Can you confirm this is unrelated?

**Comment by Erik-Lundell** [comment]
> `test-arm-reference-delegation` job is still failing but looks conv related? Can you confirm this is unrelated?

Unrelated.

**Pull request closed without merging**